### PR TITLE
Remove python2

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request adds separate provisioners to upgrade the instance and remove python2 before executing the main playbook.

## 💭 Motivation and Context

By upgrading the instance first, we ensure that everything we install is the latest.  The base AMI we are using comes with python2 preinstalled, and we only want to use python3, so it makes sense to remove it altogether.  This also stops Ansible from selecting the wrong version of python.

## 🧪 Testing

I was able to successfully build a new Kali AMI for staging with these changes.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/requirements.yml [hunk 1]
```diff
@@ -20,12 +20,18 @@
   name: nvme
 - src: https://github.com/cisagov/ansible-role-persist-journald
   name: persist_journald
+- src: https://github.com/cisagov/ansible-role-pip
+  name: pip
 - src: https://github.com/cisagov/ansible-role-python
   name: python
+  version: bugfix/install-python3-minimal
```
Comment (mcdonnnj) [inline src/requirements.yml:27]: You have a draft PR for this branch right now. Do you plan on switching to a regular PR and merging soon?

File: src/requirements.yml [hunk 2]
```diff
@@ -20,12 +20,18 @@
...
+- src: https://github.com/cisagov/ansible-role-remove-python2
+  name: remove_python2
 - src: https://github.com/cisagov/ansible-role-ufw
   name: ufw
 - src: https://github.com/cisagov/ansible-role-upgrade
   name: upgrade
+  version: bugfix/update-cache-first
```
Comment (mcdonnnj) [inline src/requirements.yml:34]: There is an active PR for this branch right now. Do you plan on merging it once it's approved?

File: src/requirements.yml (hunk 1 above)
Comment (jsf9k) [reply src/requirements.yml]: The PR has been merged, so I have switched back to the default branch.

File: src/requirements.yml (hunk 2 above)
Comment (jsf9k) [reply src/requirements.yml]: The PR has been merged, so I have switched back to the default branch.

**Pull request merged**